File: Programmers/Lv3/hanoi.cpp
#include <bits/stdc++.h>

using namespace std;

vector<vector<int>> solution(int n);
vector<vector<int>> answer;

int main() {
    int n = 3;
    solution(n);
    return 0;
}

void hanoi(int start, int end, int remainder, int n) {
    if (n == 0) return;
    hanoi(start, remainder, end, n - 1);
    answer.push_back({start, end});
    hanoi(remainder, end, start, n - 1);
}

vector<vector<int>> solution(int n) {
    hanoi(1, 3, 2, n);
    return answer;
}

File: java/src/programmers/lv3/Hanoi.java
package programmers.lv3;

import java.util.ArrayList;
import java.util.List;

public class Hanoi {
    public int[][] solution(int n) {
        List<int[]> answer = new ArrayList<>();
        hanoi(1,2,3,n, answer);

        return answer.toArray(int[][]::new);
    }
    private void hanoi(int start,int mid,int end,int n, List<int[]> answer ){
        if(n==0)return;
        hanoi(start,end,mid,n-1,answer);
        answer.add(new int[]{start,end});
        hanoi(mid,start,end,n-1,answer);
    }
}
